src/libfetchers/filtering-input-accessor.hh

```diff
@@ -64,9 +66,16 @@ struct AllowListInputAccessor : public FilteringInputAccessor
      */
     virtual void allowPrefix(CanonPath prefix) = 0;
 
+    /**
+     * Grant access to a path (but not anything underneath the path,
+     * if it's a directory).
```

Comment:
What about access to the parents? I think you accidentally wrote the wrong direction here.

The words for path containment are confusing. Maybe that happened? E.g. a path prefix is contained in a more specific path, but a more specific path contains fewer things.
Parent/child terms are unambiguous, but then I don't have a good name for the "parent or same" kind of relation. I must be forgetting something, but using parent/child is always good when they apply.